server/datastore/mysql/apple_mdm.go

```diff
@@ -862,11 +864,16 @@ func (ds *Datastore) UpdateHostTablesOnMDMUnenroll(ctx context.Context, uuid str
 			return ctxerr.Wrap(ctx, err, "getting host id from UUID")
 		}
 
-		_, err = tx.ExecContext(ctx, `
-			DELETE FROM host_mdm
+		res, err := tx.ExecContext(ctx, `
+			UPDATE host_mdm 
+			SET enrolled = 0
 			WHERE host_id = ?`, hostID)
```

Comment:
The problem is that `UpdateHostTablesOnMDMUnenroll` is called when a host is un-enrolled from Fleet, but not when it's un-enrolled from a third-party MDM, so this is probably not the right fix for this issue :(

Reply:
This issue is with turning off MDM in Fleet though. A third-party MDM shouldn't be a problem. It should just update as usual via osquery detail query.

Reply:
> This issue is with turning off MDM in Fleet though

I don't belive this is correct, but if we're sure the issue _only_ happens when turning off Fleet's MDM, then this is not a release blocker!

>  It should just update as usual via osquery detail query.

the problem is that I'll update to `installed_from_dep` to `false`, because the device is not enrolled into an MDM, from the [doc](https://fleetdm.com/tables/mdm):

> `installed_from_dep` Indicates if the MDM payload was installed via DEP or not.

that's why in the issue I suggested using `dep_capable` instead:

> `dep_capable Indicates` if the computer is DEP capable or not, even if it is not currently enrolled into MDM.

Reply:
all of that is assuming that `installed_from_dep` is playing a role here to determine if we should show the automatic enrollment modal or not, but maybe my recollection of how this works is wrong?

Reply:
> all of that is assuming that `installed_from_dep` is playing a role here to determine if we should show the automatic enrollment modal or not, but maybe my recollection of how this works is wrong?

It does play a role. I can see how this will be an problem with the work you're doing if you're planning to set the `installed_from_dep` value that way. I can expand the scope of this to make the change to `dep_capable`.

Reply:
@gillespi314 just to over-communicate, I'm not touching the `installed_from_dep` value in any way.

If adding `dep_capable` is too complex. also acceptable if there's a workaround for this release

Reply:
Gotcha. With that in mind, the bigger issue, I think, is that we're not updating the `host_mdm` entry if osquery reports nil for the MDM detail query. I'm trying some tests to confirm how an unenrolled host answers the detail query now..

Reply:
@roperzh here's what I've found:

- When we ingest the MDM detail query for a host, we delete from `host_mdm` if the MDM server url is [empty](https://github.com/fleetdm/fleet/blob/224225a8aec84389bd426f6829d39e0b43f26acb/server/datastore/mysql/hosts.go#L2653).
- Then with no `host_mdm` record, the host details API [response](https://github.com/fleetdm/fleet/blob/224225a8aec84389bd426f6829d39e0b43f26acb/server/fleet/hosts.go#L793-L794) will be `null` for `host.mdm`.

So it seems like we need to reconsider how we want to persist the host data when we can deduce an MDM unenroll event from a change in the MDM detail query results for a host. What do you think?